src/query/shorthand.ts

```diff
@@ -203,10 +232,20 @@ export function fieldDef(encQ: EncodingQuery,
     }
   }
 
-  // Scale
-  // TODO(#226):
-  // write toSpec() and toShorthand() in a way that prevents outputting inapplicable scale, sort, axis / legend
   for (const nestedPropParent of [Property.SCALE, Property.SORT, Property.AXIS, Property.LEGEND]) {
+    if ((nestedPropParent === Property.AXIS) && !isEnumSpec(encQ.channel) && !CHANNEL_SUPPORTS_AXIS[encQ.channel as Channel]) {
```

Comment:
These 4 conditions seems unnecessary redundant.  Can't you combine them to one line?